pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go

```diff
@@ -570,8 +564,16 @@ func (s *Scheduler) removeIssuer(issuerID iotago.AccountID, err error) {
 	}
 
 	s.deficits.Delete(issuerID)
+	s.basicBuffer.RemoveIssuerQueue(issuerID)
+}
 
-	s.basicBuffer.RemoveIssuer(issuerID)
+func (s *Scheduler) getOrCreateIssuer(accountID iotago.AccountID) *IssuerQueue {
+	issuerQueue, created := s.basicBuffer.GetOrCreateIssuerQueue(accountID)
+	if created {
+		s.deficits.Set(accountID, 0)
```

Comment:
Good catch, you are correct! This would have meant issuer's deficits going to 0 every time their queue went empty :o
Updating this now to use GetOrCreate method of the shrinkingmap to set the deficit only if it doesn't exist.